framework/src/outputs/TopResidualDebugOutput.C

```diff
@@ -113,6 +115,23 @@ TopResidualDebugOutput::printTopResiduals(const NumericVector<Number> & residual
     if (vec[i]._is_scalar)
       Moose::err << "(SCALAR)\n";
     else
-      Moose::err << "at node " << vec[i]._nd << '\n';
+    {
+      // Create subdomain list string for node
+      std::stringstream subdomains_ss;
+      subdomains_ss << "{";
+      bool prepend_comma = false;
+      for (const auto & subdomain_id : vec[i]._subdomain_ids)
+      {
+        if (prepend_comma)
+          subdomains_ss << ", ";
+        else
+          prepend_comma = true;
+        subdomains_ss << "'" << mesh.getSubdomainName(subdomain_id) << "'";
+      }
+      subdomains_ss << "}";
```

Comment:
This whole block of code is unneeded, use Moose::stringify() and pass in your container. It should work on this an pretty-print it for you.

Reply:
`Moose::stringify()` sort-of pretty-prints it: I get "block_one,block_two" instead of what I had before, "{'block_one', 'block_two'}". I'd argue that mine is prettier. At a minimum, for clarity in the output, I'd like containing braces for the string, so I'd still need to append those. Optionally, I think the quotes make it more clear that they are separate items, but I could live without them. Ideally, there'd be a `Moose::prettyPrintVector()` or something like this (or altering the `stringify` method, but I know that would break things). How should I proceed here?

Reply:
I agree, I'll see if anything breaks by making `stringify` "prettier".